src/core/pile.sol

```diff
@@ -128,11 +128,18 @@ contract Pile is DSNote, TitleOwned {
         beans.drip(loans_[loan].fee);
     }
 
-    // repay() a certain amount of token from the user to the Pile
-    function repay(uint loan, uint wad) public owner(loan) note {
-        // moves currency from usr to pile and reduces debt
-        require(loans_[loan].balance == 0,"before repay loan needs to be withdrawn");
+
+    // recovery used for defaulted loans_
+    function recovery(uint loan, uint wad) public auth {
+        doRepay(loan, wad);
+
+        uint loss = beans.debtOf(loan, loans_[loan].fee);
```

Comment:
this is not actually the loss but the amount we have made up from recovering the defaulted loan. the loss would be the remaining debt after the below `beans.decLoanDebt` method is called.

Reply:
@charly doRepay is calling the beans.decLoanDebt(). From my understanding... Manuel is just using the computed loss to decrease the the debt amount in beans... because no payment for this loan is expected anymore

Reply:
yes. I just did it in that way to re-use the existing `decLoanDebt` method to set the loan to `zero`